libgraph/include/katana/GraphTopology.h

```diff
@@ -994,51 +1084,70 @@ class KATANA_EXPORT UndirectedTopologyWrapper : public GraphTopologyTypes {
   // fake in-edge-ID == real in-edge-ID + out().NumEdges();
 
 public:
-  using edge_iterator =
-      katana::DisjointRangesIterator<boost::counting_iterator<Edge>>;
+  using in_out_edge_iterator = OpaqueIDCountingIterator<InOutEdgeHandle>;
+  using edge_iterator = katana::DisjointRangesIterator<
+    in_out_edge_iterator>;
   using edges_range = StandardRange<edge_iterator>;
 
+
   UndirectedTopologyWrapper(
       std::shared_ptr<const OutTopo> out,
       std::shared_ptr<const InTopo> in) noexcept
       : out_topo_(std::move(out)), in_topo_(std::move(in)) {}
 
   auto NumNodes() const noexcept { return out().NumNodes(); }
 
-  // TODO(amber): Should it be sum of in and out edges?
   auto NumEdges() const noexcept { return out().NumEdges(); }
 
   /// Gets the edge range of some node.
   ///
   /// \param node node to get the edge range of
   /// \returns iterable edge range for node.
   edges_range UndirectedEdges(Node node) const noexcept {
-    return MakeDisjointEdgesRange(out().OutEdges(node), in().OutEdges(node));
+    auto out_range_pair = std::make_pair(
+        out().RawEdgesBegin(node), out().RawEdgesEnd(node));
+    auto in_range_pair = std::make_pair(
+        in().RawEdgesBegin(node), in().RawEdgesEnd(node));
+    return MakeDisjointEdgesRange(out_range_pair, in_range_pair);
   }
 
-  Node UndirectedEdgeNeighbor(Edge eid) const noexcept {
-    if (is_in_edge(eid)) {
-      return in().OutEdgeDst(real_in_edge_id(eid));
+  Node UndirectedEdgeNeighbor(const InOutEdgeHandle& eh) const noexcept {
+    if (is_in_edge(eh)) {
+      return in().OutEdgeDst(real_edge_handle(eh));
     }
-    return out().OutEdgeDst(eid);
+    return out().OutEdgeDst(OutEdgeHandle{eh.value()});
+  }
+
+  // TODO(amber): It is possible to call UndirectedEdgeNeighbor with an
+  // OutEdgeHandle returned by Edges() below. Which leads me to suggest that we
+  // should drop UndirectedEdgeNeighbor vs OutEdgeDst and stick to a generic name
+  // like GetEdgeDest
+  Node UndirectedEdgeNeighbor(const OutEdgeHandle& eh) const noexcept {
```

Comment:
It's possible, but why do we need to add it? You can return `InOutEdgeHandle` from `Edges()`

Reply:
Yes, this comment is outdated. I ended up returning the same type of `edges_range` (i.e. InOutEdgeHandle type) and you're right, this invariant should be established by each topology/view class. Can't have some methods returning a different edge range.